cmd/pipelinelistener/README.md

```diff
@@ -0,0 +1,61 @@
+# Pipeline Listener
+
+The pipeliner listener proposal defines a process by which Tekton PipelineRuns can be directly triggered by events, which it consumes and handles to kickoff the pipeline.
+
+To do this, an optional CRD `PipelineListener` is provided. Once defined, the listener provides support for consuming CloudEvent and producing a predefined PipelineRun. It is intentionally designed to allow for other sources beyond CloudEvents.
+
+The only event-type supported by this proposal is `com.github.checksuite`.
+
+# Minikube instructions
+
+To dev/test locally with minikube:
+
+
+* Get the `ko` command: `go get -u github.com/google/ko/cmd/ko`
+* Load your docker environment vars: `eval $(minikube docker-env)`
+* Start a registry: `docker run -it -d -p 5000:5000 registry:2`
+* Set `KO_DOCKER_REPO` to local registry: `export KO_DOCKER_REPO=localhost:5000/<myproject>`
+* Apply tekton components: `ko apply -L -f config/`
+* Create a PipelineListener (such as the example below) and await cloud events.
+* Create a service or expose the pods port locally to access the endpoint.
+* Listener is configured for port `8082`.
+
+
+```
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineListener
+metadata:
+  name: test-build-pipeline-listener
+  namespace: tekton-pipelines
+spec:
+  selector:
+    matchLabels:
+      app: test-build-pipeline-listener
+  serviceName: test-build-pipeline-listener
+  template:
+    metadata:
+      labels:
+        role: test-build-pipeline-listener
+    spec:
+      serviceAccountName: tekton-pipelines-controller
+  listener-image: github.com/tektoncd/pipeline/cmd/pipelinelistener
+  event-type: com.github.checksuite
```

Comment:
Is there a way to take data out of the event payload and thread that through into the pipeline via creation of a new resource, or similar?

Reply:
Here's a strawperson I put together on how binding eventdata into a resource might work: https://gist.github.com/dlorenc/bdd3dc952cd9b9cb10b837224b031214

Any thoughts?

Reply:
I find this very interesting! Admittedly though, I am still processing how the example `EventBinding` type fits into the puzzle, inre: this PR.

Would that kind be used to swawn a listener behind the scenes based on the binding type? Or am I way off :)